src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -230,25 +206,16 @@ public async Task PartialValidationAsync(ValidationContext validationContext)
             }
         }
 
-        private async Task ExecuteRulesAsync(List<ConsensusRuleDescriptor> rules, RuleContext ruleContext)
+        private async Task ExecuteRulesAsync(IEnumerable<AsyncConsensusRule> asyncRules, RuleContext ruleContext)
         {
             try
             {
                 using (new StopwatchDisposable(o => this.PerformanceCounter.AddBlockProcessingTime(o)))
                 {
                     ruleContext.SkipValidation = ruleContext.ValidationContext.ChainTipToExtend.IsAssumedValid;
 
-                    foreach (ConsensusRuleDescriptor ruleDescriptor in rules)
-                    {
-                        if (ruleContext.SkipValidation && ruleDescriptor.RuleAttribute.CanSkipValidation)
```

Comment:
A suggestion from Fassa was to still check the can skip validation but instead we put it as a property on the rule itself so no need for attributes.